ventilator_serial.py

```diff
@@ -106,11 +126,32 @@ def run(self, name):
 
                 # handle settings
                 for msgtype in proto.settings:
+                    val = tokens[1]
+                    id = tokens[2]
                     if line.startswith((msgtype + '=')):
                         if proto.settings_values[msgtype] != val:
+                            # send to GUI
                             self.request_queue.put({'type': 'setting',
                                                     'key': msgtype,
                                                     'value': val})
+                            # acknowledge receipt
+                            self.out_queue.put({'type': proto.ack, 'val': id })
+
+
+                # resend messages waiting for ack
+                now = datetime.utcnow().timestamp()
+                delete = [] 
+                for waiting_message in waiting_for_acks.items():
+                    if waiting_message['sent_at'] + 1000 < now:
```

Comment:
`datetime.utcnow().timestamp()` returns a value in seconds so I don't think you want to wait 1000 seconds. Also you're missing the import `from datetime import datetime` for this to work I believe.